src/platform/nxp/common/NetworkCommissioningWiFiDriver.cpp

```diff
@@ -89,10 +95,14 @@ CHIP_ERROR NXPWiFiDriver::Init(NetworkStatusChangeCallback * networkStatusChange
     mpConnectCallback      = nullptr;
     mpStatusChangeCallback = networkStatusChangeCallback;
 
-    // Connect to saved network
+    // Try to connect to saved network
     err = ConnectWiFiNetwork(mSavedNetwork.ssid, ssidLen, mSavedNetwork.credentials, credentialsLen);
+    if (err != CHIP_NO_ERROR)
```

Comment:
Recommendation would be to still return an error if ConnectWiFiNetwork is failing as it would mean that a critical error happen and upper layer needs to be informed.